editor/src/components/inspector/sections/layout-section/layout-section.tsx

```diff
@@ -1,42 +1,47 @@
 import * as React from 'react'
-import { CanvasRectangle, LocalRectangle } from '../../../../core/shared/math-utils'
 import { LayoutSystemSubsection } from './layout-system-subsection/layout-system-subsection'
 import { SelfLayoutSubsection } from './self-layout-subsection/self-layout-subsection'
-import { CSSPosition } from '../../common/css-utils'
-import {
-  DetectedLayoutSystem,
-  SpecialSizeMeasurements,
-} from '../../../../core/shared/element-template'
+import { emptySpecialSizeMeasurements } from '../../../../core/shared/element-template'
 import { betterReactMemo } from '../../../../uuiui-deps'
-
-export interface ResolvedLayoutProps {
-  frame: LocalRectangle | null
-  parentFrame: CanvasRectangle | null
-}
+import { useEditorState } from '../../../editor/store/store-hook'
+import { fastForEach } from '../../../../core/shared/utils'
+import { MetadataUtils } from '../../../../core/model/element-metadata-utils'
+import { SpecialSizeMeasurementsKeepDeepEquality } from '../../../editor/store/store-deep-equality-instances'
 
 interface LayoutSectionProps {
-  input: ResolvedLayoutProps
-  specialSizeMeasurements: SpecialSizeMeasurements
-  isChildOfFlexComponent: boolean
-  position: CSSPosition | null
   hasNonDefaultPositionAttributes: boolean
-  parentFlexAxis: 'horizontal' | 'vertical' | null
   aspectRatioLocked: boolean
   toggleAspectRatioLock: () => void
 }
 
 export const LayoutSection = betterReactMemo('LayoutSection', (props: LayoutSectionProps) => {
+  const specialSizeMeasurements = useEditorState(
+    (state) => {
+      let foundSpecialSizeMeasurements = emptySpecialSizeMeasurements
+      fastForEach(state.editor.selectedViews, (path) => {
```

Comment:
AFAIK the rest of the inspector uses a target path property that is independent of selection – we could use that here too, or at least put a comment here noting the divergence!